File: BruteForce/7453/7453.py
n=int(input())
arr_a=[]
arr_b=[]
arr_c=[]
arr_d=[]
for _ in range(n):
    [a,b,c,d]=list(map(int,input().split()))
    arr_a.append(a)
    arr_b.append(b)
    arr_c.append(c)
    arr_d.append(d)

arr1=[]
for i in range(n): #a와 b의 합 배열을 저장
    for j in range(n):
        sumAB=arr_a[i]+arr_b[j]
        arr1.append(sumAB)

arr2=[]

for i in range(n): #c와 d의 합 배열을 저장
    for j in range(n):
        sumCD=arr_c[i]+arr_d[j]
        arr2.append(sumCD)
arr2.sort() #이분 탐색을 위해 정렬

ans=0
for i in range(len(arr1)): #arr1의 배열을 돌면서
    num=arr1[i]
    start=0
    end=len(arr2)

    while(start<=end): #이분탐색으로 합이 0이 되는 갯수를 찾기
        mid=(start+end)//2
      
        if(arr2[mid]==-num):
            ans+=1
            break
        elif(arr2[mid]<-num):
          start=mid+1
        
        else:
          end=mid-1
            
print(ans)

File: BruteForce/7453/7453.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int main() {
	int n;
	cin >>n;
	vector <long long> a(n);
	vector <long long> b(n);
	vector <long long> c(n);
	vector <long long> d(n);
	
	for (int i=0; i<n;i++) {
		scanf("%ld %ld %ld %ld",&a[i],&b[i],&c[i],&d[i]);
	}
	vector <long long> arr1;
	vector <long long> arr2;
	for (int i=0; i<n;i++) {
		for (int j=0; j<n;j++) {
			arr1.push_back(a[i]+b[j]);
			arr2.push_back(c[i]+d[j]);
		}
	}


	sort(arr2.begin(),arr2.end());
	
	int ans=0;
	
	for (int i=0; i<arr1.size();i++) {
		long long num=arr1[i];
		int cnt=upper_bound(arr2.begin(), arr2.end(),-num)-lower_bound(arr2.begin(), arr2.end(),-num);
		if(cnt>0){
		    ans+=cnt;
		}
	
	}
	cout <<ans<<'\n';
	return 0;
}